tests/helpers/daemon.rs

```diff
@@ -0,0 +1,130 @@
+extern crate tempdir;
+
+use self::tempdir::TempDir;
+use super::mpd;
+use std::fs::{File, create_dir};
+use std::io::{Write, Read};
+use std::os::unix::net::UnixStream;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Child, Stdio};
+
+struct MpdConfig {
+    db_file: PathBuf,
+    music_directory: PathBuf,
+    playlist_directory: PathBuf,
+    config_path: PathBuf,
+    sock_path: PathBuf,
+}
+
+impl MpdConfig {
+    pub fn new<P>(base: P) -> MpdConfig
+        where P: AsRef<Path>
+    {
+        let base = base.as_ref();
+        MpdConfig {
+            db_file: base.join("db"),
+            music_directory: base.join("music"),
+            playlist_directory: base.join("playlists"),
+            config_path: base.join("config"),
+            sock_path: base.join("sock"),
+        }
+    }
+
+    fn config_text(&self) -> String {
+        format!(r#"
```

Comment:
I don't see a problem here. Now you can see the generated text immediately in the place it is generated. If we move this string to some other place, we will have to look it up to understand the function output better.